### PR TITLE
Continue rewriting workflows to reusable tests

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
A follow-up for #56385 

#### TODO

- [x] Add validation for CI_CONFIG.build_configs if the name is equal to the dict's key
- [x] Write a script to validate `.github/workflows/reusable_*` files. The idea is about `GITHUB_JOB_OVERRIDDEN` and maybe something more
- [x] adjust build_report_check to a reusable_test. It requires some logic change
- [x] Skip unnecessary `PrepareStrategy` step when `batches: 1`
- [ ] rewrite cherry_pick.yml, but not necessary

### Review
To cover each workflow and avoid typos or stupid mistakes, I'd like to have 4 additional eyes there. If you are the second one who reviews it, please mark the line as completed and add your GH login.

*What to catch:* a) if the number of batches remains the same; b) if the `test_name` is the same as the `$CHECK_NAME` used to be; c) if the `runner_type` remains the same, as used to be (exceptions: there was a different type for StressTestTsan, it's irrelevant now); d) the `run_command` should be the same as before.

- [ ]  .github/workflows/backport_branches.yml
- [ ]  .github/workflows/docs_check.yml
- [ ] .github/workflows/jepsen.yml
- [ ] .github/workflows/libfuzzer.yml
- [ ] .github/workflows/master.yml
- [x] .github/workflows/nightly.yml - a trivial change
- [ ] .github/workflows/pull_request.yml
- [ ] .github/workflows/release_branches.yml
- [x] .github/workflows/woboq.yml - just deleted

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 24e55320629b960d711766683d83be856bd8fc77 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56501/24e55320629b960d711766683d83be856bd8fc77/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6810173780">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56501/24e55320629b960d711766683d83be856bd8fc77/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: .github/workflows/reusable_test.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  PrepareStrategy:
+    if: ${{inputs.batches > 0}}  # batches < 1 is misconfiguration
+    runs-on: [self-hosted, style-checker-aarch64]
+    outputs:
+      batches: ${{steps.batches.outputs.batches}}
+    steps:
+      - name: Calculate batches
+        id: batches
+        run: |
+          batches_output=$(python3 -c 'import json; print(json.dumps(list(range(${{inputs.batches}}))))')
```
Comment (mkaynov) [inline .github/workflows/reusable_test.yml:51]: I believe we can avoid using separate runner for this

File: .github/workflows/reusable_test.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        with:
+          path: ${{ env.REPORTS_PATH }}
+      - name: Setup batch
+        if: ${{ inputs.batches > 1}}
+        run: |
+          cat >> "$GITHUB_ENV" << 'EOF'
+          RUN_BY_HASH_NUM=${{matrix.batch}}
+          RUN_BY_HASH_TOTAL=${{inputs.batches}}
+          EOF
+      - name: Build
```
Comment (mkaynov) [inline .github/workflows/reusable_test.yml:89]: why is it called Build?

File: .github/workflows/reusable_test.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        run: ${{inputs.run_command}}
+      - name: Upload build URLs to artifacts
```
Comment (mkaynov) [inline .github/workflows/reusable_test.yml:91]: what kind of artifacts is uploaded here?

File: .github/workflows/reusable_test.yml (hunk 1 above)
Comment (Felixoid) [reply .github/workflows/reusable_test.yml:51]: Nope, can't. You are free to fork https://github.com/Felixoid/actions-experiments/commit/69e917383d4e0bf5f44b4cdb8987e490b3e22651 and play, it's the minimum reproducible thing

File: .github/workflows/reusable_test.yml (hunk 2 above)
Comment (Felixoid) [reply .github/workflows/reusable_test.yml:89]: Leftover, will eventually go

File: .github/workflows/reusable_test.yml (hunk 3 above)
Comment (Felixoid) [reply .github/workflows/reusable_test.yml:91]: Same, it's trash

File: .github/workflows/reusable_test.yml (hunk 1 above)
Comment (Felixoid) [reply .github/workflows/reusable_test.yml:51]: although, we could try to output it from JS action. It could work out

Comment (Felixoid) [reply .github/workflows/reusable_test.yml:51]: But no, even our [debug](https://github.com/ClickHouse/ClickHouse/blob/master/.github/workflows/debug.yml) launches "somewhere"

Comment (Felixoid) [reply .github/workflows/reusable_test.yml:51]: BTW, I thought it will be used to calculate if it's necessarily to run the following tests :wink:

**Comment by alesapin** [comment]
image

**Comment by Felixoid** [comment]
Every type of the test is finished, LET'S GOOOO!

**Pull request merged**